test/test_arrays.jl

```diff
@@ -0,0 +1,122 @@
+import SparseConnectivityTracer as SCT
+using SparseConnectivityTracer: GradientTracer
+using LinearAlgebra: Symmetric, Diagonal
+using LinearAlgebra: det, logdet, logabsdet, norm, opnorm
+using LinearAlgebra: eigen, eigmax, eigmin
+using LinearAlgebra: inv, pinv
+using SparseArrays: sparse, spdiagm
+using Test
+
+PATTERN_FUNCTIONS = (connectivity_pattern, jacobian_pattern, hessian_pattern)
+
+TEST_SQUARE_MATRICES = Dict(
+    "`Matrix` (3×3)" => rand(3, 3),
+    "`Symmetric` (3×3)" => Symmetric(rand(3, 3)),
+    "`Diagonal` (3×3)" => Diagonal(rand(3)),
+)
+TEST_MATRICES = merge(TEST_SQUARE_MATRICES, Dict("`Matrix` (3×4)" => rand(3, 4)))
```

Comment:
add sparse test matrices?

Reply:
Otherwise we can't test that the fixes you introduce actually solve #108

Reply:
No, that's what the closures in the tests are currently for.